backfill.py

```diff
@@ -130,27 +148,126 @@ def get_result_from_durations(start_up_durations_path, analyzed_path):
+
+    if assemble_proc.returncode != 0:
+        print(("\n\nSomething went wrong while assembling this build: {build} . The associated error message was:"
+               "\n\n {error}".format(build=build_type, error=checkout_proc.stderr.decode('utf-8').strip("\n"))),
+              file=sys.stderr)
+
+
+def build_apk_path_string(repository_path, build_type, phone_architecture):
+    apk_name = "app-{phone_arch}-{build_type}.apk".format(phone_arch=phone_architecture, build_type=build_type)
+    build_apk_destination = os.path.join(repository_path, "app", "build", "outputs", "apk", build_type, apk_name)
+    return build_apk_destination
+
+
+def move_apk_to_cwd(apk_path, commit_hash):
+    new_apk_name = "apk_commit_" + commit_hash + ".apk"
+    proc = subprocess.run(["mv", apk_path, new_apk_name])
+    if proc.returncode != 0:
+        print(("\n\nSomething went wrong while moving the built apk: {apk} . The associated error message was:"
+               "\n\n {error}".format(apk=apk_path, error=proc.stderr.decode('utf-8').strip("\n"))),
+              file=sys.stderr)
+    return new_apk_name
+
+
+def build_apks_for_commits(
+        start_commit=None, end_commit=None, repository_path=None,
+        build_type=None, architecture=None, remote_name=""):
+    apk_metadata_array = []
+
+    fetch_repository(repository_path, remote_name)
+    array_of_commit_hash = get_all_commits_in_commits_range(start_commit, end_commit, repository_path)
+    for commit in array_of_commit_hash:
+        build_apk_for_commit(commit, repository_path, build_type)
+        built_apk_name = build_apk_path_string(repository_path, build_type, architecture)
+        new_apk_name = move_apk_to_cwd(built_apk_name, commit)
+        apk_metadata_array.append({
+            KEY_NAME: new_apk_name,
+            KEY_DATETIME: "",
+            KEY_COMMIT: commit,
+            KEY_ARCHITECTURE: architecture})
+    return apk_metadata_array
 
 
 def cleanup(array_of_apk_path):
     for i in array_of_apk_path:
         subprocess.run(["rm", i[KEY_NAME]])
 
 
-def main():
-    args = parse_args()
-    if args.type == "commits" and args.fenix_path is None:
+def validate_args(args):
+    if args.build_source == "commitsRange" and args.repository_to_test_path is None:
         raise Exception("Provide the path to your fenix repository to run this script with the commits option")
+    if args.build_source == "commitsRange" and not args.startcommit and not args.endcommit:
+        raise Exception("Running backfill with commits between two commits requires a start and end commit")
+    if args.build_source == "commitsRange" and not args.repository_to_test_path:
+        raise Exception("Running backfill with any commits option " +
+                        "requires a path to a repository where git can be used")
+    if args.build_source == "commitsRange" and not args.release_channel:
```

Comment:
nitpick (blocking): this is no longer necessary that `release_channel` is a positional argument.